source/overview/implementation_plan.rst

```diff
@@ -0,0 +1,369 @@
+Implementation Plan
+================================
+
+1   Introduction
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1.1 Purpose
+.........................................
+
+Mattermost is an open source, private cloud alternative to proprietary SaaS messaging solutions designed to offer organizations superior control, data sovereignty, configurability, freedom from lock-in, and enhanced security by keeping vital communications behind your firewall. You have chosen to implement Mattermost in our supported `editions:`_
+
+.. _`editions:`: https://about.mattermost.com/features/
+
+ - Team Edition
+
+ - Enterprise Edition (E10) which adds additional features to the platform including AD/LDAP integration, multi-factor authentication ("MFA"), custom branding, advanced access control policy and next business day support.
+
+ - Enterprise Edition (E20) includes E10 features plus SAML 2.0 authentication, security controls, compliance reporting, high availability, and enterprise class support. 
+
+The purpose of this Implementation Plan is to:
+
+ - Describe the solution to be deployed;
+
+ - Outline the process for implementing the solution (key personnel, tasks, schedule, etc.);
+
+ - Identify support required during the implementation process.
+
+2 System Overview
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This section provides a brief overview of the system to be implemented, including a description of the system and its organization.
+
+2.1 System Description
+.........................................
+
+Mattermost is a messaging solution designed to remove friction for inter-organizational communications. You are implementing Mattermost in order to:
+
+1. **[STATE REASON]**
+
+2. **[STATE REASON]**
+
+3. **[STATE REASON]**
+
+In this implementation, you are planning to deploy Mattermost to **[NUMBER OF USERS]** users on the following teams:
+
+1. **[TEAM 1]**
+
+2. **[TEAM 2]**
+
+3. **[TEAM 3]**
+
+
+2.2  System Organization
+.........................................
+
+A Mattermost Enterprise E20 implementation consists of the following system components:
+
+ - **Mattermost Server**
+   - The Mattermost Server is a single binary that includes the RESTful JSON web service, authentication client, authentication provider, notification service, and data management service. The Mattermost Server can be deployed in stand-alone or high availability mode where two or more servers are clustered together using gossip protocol and a proxy server that routes traffic from client applications to healthy servers in the cluster.
+ - **Mattermost Database**
+   - A MySQL or PostgreSQL database in stand-alone or high-availability configurations (master with read replicas). When choosing the database, consider the `inherent search limitations of the database type <https://docs.mattermost.com/install/requirements.html#database-software>`_ and whether `Elasticsearch <https://docs.mattermost.com/deployment/elasticsearch.html>`_ is most appropriate for your deployment.
```

Comment:
@james-mm Added a brief note about the inherent search limitations of the database type, and that Elasticsearch may sometimes be the most appropriate option for E20 customers.